### ISSUE TITLE
Improve `Style/OptionalBooleanParameter` cop

### ISSUE THREAD

**Issue by ydakuka** [rubocop/rubocop#12347]
## Description

Use keyword arguments instead of option hashes.

https://rubystyle.guide/#keyword-arguments-vs-option-hashes

https://rubystyle.guide/#boolean-keyword-arguments

## Behavior

```ruby
# bad
def some_method(options = {})
  bar = options.fetch(:bar, false)
  puts bar
end

# good
def some_method(bar: false)
  puts bar
end
```

**Comment by koic** [comment]
> ```ruby
> # bad
> def self.call(options = {})
>   new(options).call
> end
>
> # good
> def self.call(...)
>   new(...).call
> end
> ```

`**options` and `*args` are different, I think. Is there a basis for this case in the style guide?

**Comment by ydakuka** [comment]
> **options and *args are different, I think.

No, arguments forwarding handles both of these cases.

> Is there a basis for this case in the style guide?

Yes, see [the rule](https://rubystyle.guide/#arguments-forwarding) and [Style/ArgumentsForwarding  cop](https://docs.rubocop.org/rubocop/cops_style.html#styleargumentsforwarding).

**Comment by koic** [comment]
I see my explanation was unclear. `options = {}` doesn't seem to correspond to `**options`, meaning it doesn't appear to be equivalent to `...`.
That is, instead of `call(k: v)`, it allows calls like `call(x, y)` (not hash arguments style), which changes the meaning.

**Comment by ydakuka** [comment]
Thanks for the explanation, I've removed the example from the issue description.

**Comment by fatkodima** [comment]
https://rubystyle.guide/#boolean-keyword-arguments is already implemented via `Style/OptionalBooleanParameter`

**Comment by ydakuka** [comment]
> https://rubystyle.guide/#boolean-keyword-arguments is already implemented via Style/OptionalBooleanParameter

Yes, but

1) the rule is duplicated in the style guide (here https://rubystyle.guide/#keyword-arguments-vs-option-hashes and here https://rubystyle.guide/#boolean-keyword-arguments).
2) I've checked it again and it detected no offences in my snippet.

I have the code:
```ruby
class Test
  def some_method(options = {})
    bar = options.fetch(:bar, false)
    puts bar
  end
end
```

Terminal:
```bash
ydakuka@yauhenid:~/Work/main_app$ bin/rails_docker rubocop app/models/test.rb
Inspecting 1 file
.

1 file inspected, no offenses detected
```
```bash
ydakuka@yauhenid:~/Work/project$ bin/rails_docker rubocop app/models/test.rb --only Style/OptionalBooleanParameter
Inspecting 1 file
.

1 file inspected, no offenses detected
```
